File: typescript-angular/API/Client/CloudRoleApi.ts
/// <reference path="api.d.ts" />

/* tslint:disable:no-unused-variable member-ordering */

namespace API.Client {
    'use strict';

    export class CloudRoleApi {
        protected basePath = 'http://api.cloudboost.io/';
        public defaultHeaders : any = {};

        static $inject: string[] = ['$http', '$httpParamSerializer'];

        constructor(protected $http: ng.IHttpService, protected $httpParamSerializer?: (d: any) => any, basePath?: string) {
            if (basePath) {
                this.basePath = basePath;
            }
        }

        private extendObj<T1,T2>(objA: T1, objB: T2) {
            for(let key in objB){
                if(objB.hasOwnProperty(key)){
                    objA[key] = objB[key];
                }
            }
            return <T1&T2>objA;
        }

        /**
         * create a new cloudrole
         * 
         * @param appId ID of app
         * @param body request body
         */
        public createRole (appId: string, body: RoleBody, extraHttpRequestParams?: any ) : ng.IHttpPromise<CloudRole> {
            const localVarPath = this.basePath + 'data/{app_id}/Role'
                .replace('{' + 'app_id' + '}', String(appId));

            let queryParameters: any = {};
            let headerParams: any = this.extendObj({}, this.defaultHeaders);
            // verify required parameter 'appId' is set
            if (!appId) {
                throw new Error('Missing required parameter appId when calling createRole');
            }
            // verify required parameter 'body' is set
            if (!body) {
                throw new Error('Missing required parameter body when calling createRole');
            }
            let httpRequestParams: any = {
                method: 'PUT',
                url: localVarPath,
                json: true,
                data: body,
                
                
                params: queryParameters,
                headers: headerParams
            };

            if (extraHttpRequestParams) {
                httpRequestParams = this.extendObj(httpRequestParams, extraHttpRequestParams);
            }

            return this.$http(httpRequestParams);
        }
    }
}

File: akka-scala/src/main/scala/io/swagger/client/api/CloudRoleApi.scala
package io.swagger.client.api

import io.swagger.client.model.CloudRole
import io.swagger.client.model.RoleBody
import io.swagger.client.core._
import io.swagger.client.core.CollectionFormats._
import io.swagger.client.core.ApiKeyLocations._

object CloudRoleApi {

  /**
   * 
   * Expected answers:
   *   code 200 : CloudRole (successful operation)
   * 
   * @param appId ID of app
   * @param body request body
   */
  def createRole(appId: String, body: RoleBody): ApiRequest[CloudRole] =
    ApiRequest[CloudRole](ApiMethods.PUT, "http://api.cloudboost.io/", "data/{app_id}/Role", "application/json")
      .withBody(body)
      .withPathParam("app_id", appId)
      .withSuccessResponse[CloudRole](200)
      


}

